[Use Cases] add additional use cases for Key in libelektra-core (#4838)
ElektraInitiative/libelektra: doc/usecases/core/UC_key_copy.md

# Use Case: `Key` copy

## Summary

- **Scope:** `libelektra-core`
- **Level:** Developer Goal
- **Actors:** Core, Caller
- **Brief:** Copying one or more fields from one `Key` to another
- **Status:** Implemented

## Scenarios

- **Precondition:**
  - [`Key` K1 has been created](UC_key_create.md)
  - [`Key` K2 has been created](UC_key_create.md)
- **Main success scenario:**

>   - Caller requests to copy one or multiple fields from `Key` k1 to `Key` k2
>   - Fields that can be copied are name, value and metadata
>   - Fields, that caller wanted to copy, of `Key` k2 now have the value of the respective fields of `Key` k1

Reviewer: ```suggestion
  - Caller requests to copy some or all data (name, value, metadata) from `Key` k1 to `Key` k2
  - The data, that caller wanted to copy, of `Key` k2 now have the value of the respective fields of `Key` k1
```
